connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteThreadRunner.scala

```diff
@@ -92,12 +92,15 @@ private[connect] class ExecuteThreadRunner(executeHolder: ExecuteHolder) extends
         "execute",
         executeHolder.responseObserver,
         executeHolder.sessionHolder.userId,
-        executeHolder.sessionHolder.sessionId)
+        executeHolder.sessionHolder.sessionId,
+        Some(executeHolder.events))
     }
   }
 
   // Inner executeInternal is wrapped by execute() for error handling.
   private def executeInternal() = {
+    // start event must be received before a cancel event from interrupt
+    executeHolder.events.postStarted()
```

Comment:
if it fails to start a thread you may end up getting an error event thrown before.
Comparing to Thriftserver, the started event there is outside the thread in SparkExecuteStatementOperation.runInternal

Reply:
Moved to SparkConnectExecutePlanHandler